releasenotes/notesRelease/38597.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+issue:
+  - 38597
+releaseNotes:
+- |
+  **Fixed** Renamed DefaultKubernetesDomain to DefaultClusterLocalDomain as mentioned TODO in the istio/pkg/config/constants/constants.go
```

Comment:
Don't need a release note for this ... just add a `release-notes-none` label to the PR to get around the check.